include/alpaka/queue/sycl/QueueGenericSyclBase.hpp

```diff
@@ -0,0 +1,252 @@
+/* Copyright 2022 Jan Stephan
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/dev/Traits.hpp>
+#include <alpaka/event/Traits.hpp>
+#include <alpaka/queue/Traits.hpp>
+#include <alpaka/wait/Traits.hpp>
+
+#include <CL/sycl.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <mutex>
+#include <shared_mutex>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace alpaka::experimental::detail
+{
+    template<typename T, typename = void>
+    inline constexpr auto is_sycl_task = false;
+
+    template<typename T>
+    inline constexpr auto is_sycl_task<T, std::void_t<decltype(T::is_sycl_task)>> = true;
+
+    template<typename T, typename = void>
+    inline constexpr auto is_sycl_kernel = false;
+
+    template<typename T>
+    inline constexpr auto is_sycl_kernel<T, std::void_t<decltype(T::is_sycl_kernel)>> = true;
+
+    class QueueGenericSyclImpl
+    {
+    public:
+        QueueGenericSyclImpl(sycl::context context, sycl::device device)
+            : m_queue{
+                std::move(context), // This is important. In SYCL a device can belong to multiple contexts.
+                std::move(device),
+                {sycl::property::queue::enable_profiling{}, sycl::property::queue::in_order{}}}
+        {
+        }
+
+        // This class will only exist as a pointer. We don't care about copy and move semantics.
+        QueueGenericSyclImpl(QueueGenericSyclImpl const& other) = delete;
+        auto operator=(QueueGenericSyclImpl const& rhs) -> QueueGenericSyclImpl& = delete;
+
+        QueueGenericSyclImpl(QueueGenericSyclImpl&& other) noexcept = delete;
+        auto operator=(QueueGenericSyclImpl&& rhs) noexcept -> QueueGenericSyclImpl& = delete;
+
+        ~QueueGenericSyclImpl()
+        {
+            try
+            {
+                m_queue.wait_and_throw();
+            }
+            catch(sycl::exception const& err)
+            {
+                std::cerr << "Caught SYCL exception while destructing a SYCL queue: " << err.what() << std::endl;
+            }
+            catch(std::runtime_error const& err)
+            {
+                std::cerr << "The following runtime errors occured while destructing a SYCL queue:" << std::endl;
+                std::cerr << err.what() << std::endl;
+            }
```

Comment:
```suggestion
            catch(sycl::exception const& err)
            {
                std::cerr << "Caught SYCL exception while destructing a SYCL queue: " << err.what() << " " << err.get_cl_code() << std::endl;
            }
            catch(std::exception const& err)
            {
                std::cerr << "The following runtime errors occured while destructing a SYCL queue:\n" << err.what() << std::endl;
            }
```
Consider adding the error code when you catch a SYCL exception. And for the second catch block please catch `std::exception` since it's at the top of the exception hierarchy.